azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunner.java

```diff
@@ -879,12 +904,47 @@ private void prepareJobProperties(final ExecutableNode node) throws IOException
       props = jobSource;
     }
 
+    if (this.azkabanProps.getBoolean(
+        ConfigurationKeys.EXECUTOR_PROPS_RESOLVE_OVERRIDE_EXISTING_ENABLED, false)) {
+      // Flow override props are configured to also override existing job props
+      // =>
+      // 5. If there are any runtime flow overrides, we apply them now.
+      final Map<String, String> flowParam =
+          this.flow.getExecutionOptions().getFlowParameters();
+      if (flowParam != null && !flowParam.isEmpty()) {
+        props.putAll(flowParam);
+      }
```

Comment:
With this block of code I think we're mixing the ability to overwrite properties with the concept of flow and job level properties.

-By providing runtime properties we are overwriting the value of already defined properties (or adding new ones).
-That should NOT interfere with the notion of flow and job level properties. I believe flow level properties are meant to define properties to be applied to the entire DAG and as a side effect offer a convenient way of defining job level properties in one place that will be applied to all jobs. However, job level properties take precedence over the same properties defined at flow level.

I don't think it's ok to add this piece of code because we're saying that flow level overwrites take precedence over the existing hierarchy of flow/job level properties. Flow level overwrites should only take precedence over flow level definition properties.

With the "job specific flow overwrites" feature added by this PR we're adding the missing piece in the puzzle which is the ability to overwrite job level definition properties.

Reply:
@ypadron-in this code block was actually added in https://github.com/azkaban/azkaban/pull/2632, which was just merged yesterday. This PR in turn, was waiting to be rebased as soon as that one is merged, as mentioned in comment https://github.com/azkaban/azkaban/pull/2597#issuecomment-729496172 – but I had not rebased before your comment here. So the block that you comment vanishes from this PR as soon as I rebase.

Reply:
Then I think we should revert that PR. Do you understand the issue there?

Reply:
Let's move this discussion to that PR -> https://github.com/azkaban/azkaban/pull/2632/files#r568464904